data/data/aws/cluster/route53/base.tf

```diff
@@ -13,6 +13,8 @@ data "aws_route53_zone" "public" {
   count = local.public_endpoints ? 1 : 0
 
   name = var.base_domain
+
+  depends_on = [aws_route53_record.api_internal_alias, aws_route53_record.api_internal_cname]
```

Comment:
I think that we should put the `depends_on` on the resource for the external records themselves. And it should be the records for the api name not the api-int name.
`aws_route53_record.api_external_alias` -> `aws_route53_record.api_external_internal_zone_alias`
`aws_route53_record.api_external_cname` -> `aws_route53_record.api_external_internal_zone_cname`

Reply:
Fixed.